### PR TITLE
Forward request id to the ratelimit service

### PR DESCRIPTION
**Pull request by junr03**

### CODE REVIEW

File: source/common/filter/ratelimit.cc [hunk 1]
```diff
@@ -33,7 +33,7 @@ Network::FilterStatus Instance::onData(Buffer::Instance&) {
     config_->stats().active_.inc();
     config_->stats().total_.inc();
     calling_limit_ = true;
-    client_->limit(*this, config_->domain(), config_->descriptors());
+    client_->limit(*this, config_->domain(), config_->descriptors(), "");
```
Comment (mattklein123) [inline source/common/filter/ratelimit.cc]: EMPTY_STRING

**Comment by mattklein123** [comment]
I see no test for the HTTP filter where you set the request ID and make sure it gets passed through. Also no test in the rate limit service itself to make sure the header gets set.

File: source/common/filter/ratelimit.cc (hunk 1 above)
Comment (junr03) [inline source/common/filter/ratelimit.cc]: changes all "" to EMPTY_STRING.

File: source/common/ratelimit/ratelimit_impl.cc [hunk 2]
```diff
@@ -50,6 +57,7 @@ void GrpcClientImpl::onSuccess() {
 
   callbacks_->complete(status);
   callbacks_ = nullptr;
+  request_id_.clear();
```
Comment (junr03) [inline source/common/ratelimit/ratelimit_impl.cc]: added

File: source/common/ratelimit/ratelimit_impl.cc [hunk 3]
```diff
@@ -32,15 +33,21 @@ void GrpcClientImpl::createRequest(pb::lyft::ratelimit::RateLimitRequest& reques
...
   callbacks_ = &callbacks;
+  request_id_ = request_id;
 
   pb::lyft::ratelimit::RateLimitRequest request;
   createRequest(request, domain, descriptors);
   service_.ShouldRateLimit(nullptr, &request, &response_, nullptr);
 }
 
+void GrpcClientImpl::onPreRequestCustomizeHeaders(Http::HeaderMap& headers) {
+  headers.addViaCopy(Http::Headers::get().RequestId, request_id_);
```
Comment (mattklein123) [inline source/common/ratelimit/ratelimit_impl.cc]: Only set if request_id_ is not empty

Comment (mattklein123) [inline source/common/ratelimit/ratelimit_impl.cc]: and add test for this case

File: source/common/ratelimit/ratelimit_impl.cc [hunk 4]
```diff
@@ -1,3 +1,4 @@
+#include <common/common/empty_string.h>
```
Comment (RomanDzhabarov) [inline source/common/ratelimit/ratelimit_impl.cc]: s/</"

File: source/common/ratelimit/ratelimit_impl.cc [hunk 5]
```diff
@@ -32,15 +34,23 @@ void GrpcClientImpl::createRequest(pb::lyft::ratelimit::RateLimitRequest& reques
...
   callbacks_ = &callbacks;
+  request_id_ = request_id;
 
   pb::lyft::ratelimit::RateLimitRequest request;
   createRequest(request, domain, descriptors);
   service_.ShouldRateLimit(nullptr, &request, &response_, nullptr);
 }
 
+void GrpcClientImpl::onPreRequestCustomizeHeaders(Http::HeaderMap& headers) {
+  if (request_id_ != EMPTY_STRING) {
```
Comment (mattklein123) [inline source/common/ratelimit/ratelimit_impl.cc]: nit: !request_id_.empty()

Comment (junr03) [inline source/common/ratelimit/ratelimit_impl.cc]: fixed

**Comment by junr03** [comment]
🚀 ✅

**Comment by RomanDzhabarov** [comment]
🚀 ✅ does not work here :-)

**Pull request merged**